From: Florian Kriener <florian@kriener.org>
To: submit@bugs.debian.org
Subject: [pm-utils] wrong path in intel-audio-powersave (and a small bug)
Date: Sat, 25 Sep 2010 11:27:30 +0200

In the script intel-audio-powersave is this loop

for dev in /sys/module/snd_*/parameters/power_save; do
	[ -w "$dev/parameters/power_save" ] || continue
     printf "Setting power savings for $s to %d..." "$dev##*/" "$1"
	echo $1 > "$dev/parameters/power_save" && echo Done. || echo Failed.
done

I think it should be

for dev in /sys/module/snd_*; do
	[ -w "$dev/parameters/power_save" ] || continue
     printf "Setting power savings for %s to %d..." "${dev##*/}" "$1"
	echo $1 > "$dev/parameters/power_save" && echo Done. || echo Failed.
done


This fixes the two bugs.

diff --git a/pm/power.d/intel-audio-powersave b/pm/power.d/intel-audio-powersave
index 36675a8..da63e40 100644
--- a/pm/power.d/intel-audio-powersave
+++ b/pm/power.d/intel-audio-powersave
@@ -20,9 +20,9 @@ EOF
 
 audio_powersave() {
     [ "$INTEL_AUDIO_POWERSAVE" = "true" ] || exit $NA
-    for dev in /sys/module/snd_*/parameters/power_save; do
+    for dev in /sys/module/snd_*; do
 	[ -w "$dev/parameters/power_save" ] || continue
-	printf "Setting power savings for $s to %d..." "$dev##*/" "$1"
+	printf "Setting power savings for %s to %d..." "${dev##*/}" "$1"
 	echo $1 > "$dev/parameters/power_save" && echo Done. || echo Failed.
     done
 }
